mantle/kola/tests/misc/zincati-deadend.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package misc
+
+import (
+	"fmt"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform/conf"
+	"github.com/vincent-petithory/dataurl"
+)
+
+var baseurl = []byte(`[cincinnati]
+base_url = "http://127.0.0.1:9001"`)
+
+var updates = []byte(`updates.enabled = true`)
+
+var graph = []byte(`{
+	"nodes": [
+	  {
+		"version": "${VERSION}",
+		"metadata": {
+		  "org.fedoraproject.coreos.scheme": "checksum",
+		  "org.fedoraproject.coreos.releases.age_index": "0",
+		  "org.fedoraproject.coreos.updates.deadend_reason": "https://github.com/coreos/fedora-coreos-tracker/issues/215",
+		  "org.fedoraproject.coreos.updates.deadend": "true"
+		},
+		"payload": "${PAYLOAD}"
+	  }
+	],
+	"edges": []
+  }`)
+
+var script = []byte(`#!/bin/bash
+  VERSION="$(/usr/bin/rpm-ostree status --json | jq '.deployments[0].version' -r)"
+  PAYLOAD="$(/usr/bin/rpm-ostree status --json | jq '.deployments[0].checksum' -r)"
+  sed -i 's/\"version\":.*/\"version\": \"'$VERSION'\",/g' /var/www/html/v1/graph
+  sed -i 's/\"payload\":.*/\"payload\": \"'$PAYLOAD'\"/g' /var/www/html/v1/graph`)
+
+var zincatidropin = []byte(`[Service]
+ExecStartPre=sleep 5
+Environment=ZINCATI_VERBOSITY="-vvvv"`)
+
+var nginxdropin = []byte(`[Unit]
+Description=nginx Server
+After=network-online.target
+Requires=network-online.target
+Before=zincati.service
+[Service]
+Restart=on-failure
+ExecStartPre=-/usr/bin/podman pull docker.io/nginx:latest
```

Comment:
I think you need `Type=oneshot` here and `RemainAfterExit=yes`. That alone should help with the race against Zincati, but if not you can also add a second `ExecStart` which e.g. keeps trying to `curl` the graph until it succeeds.